app/assets/src/components/views/amr_heatmap/AMRHeatmapVis.jsx

```diff
@@ -20,15 +33,34 @@ export default class AMRHeatmapVis extends React.Component {
     const [sampleLabels, geneLabels, alleleLabels] = this.extractLabels(
       samplesWithAMRCounts
     );
+    const alleleToGeneMap = this.mapAllelesToGenes(samplesWithAMRCounts);
     this.setState({
       sampleLabels,
       geneLabels,
       alleleLabels,
+      alleleToGeneMap,
     });
   }
 
-  componentDidUpdate() {
-    this.updateHeatmap();
+  componentDidUpdate(prevProps) {
+    if (this.props !== prevProps || this.heatmap === null) {
+      this.updateHeatmap();
+    }
+  }
+
+  // Sometimes, when presenting the tooltip popup as a user hovers over
+  // a node on the heatmap, they will be over a node where the row is
+  // an allele and the column is a sample with no AMR count for the allele.
+  // With no AMR count, there's no easy way to grab the name of the gene
+  // for the allele. Hence the allele-to-gene mapping.
```

Comment:
This makes sense! But I'm a little confused why there isn't also a gene-to-allele mapping. Isn't it possible that the sample has no AMR count for the gene as well?

Reply:
Wait never mind. If there's no AMR count for the gene, there's no way to tell which allele it is.

Reply:
Exactly!